src/software/ai/hl/stp/play/defense/defense_play_fsm.cpp

```diff
@@ -75,11 +122,43 @@ void DefensePlayFSM::defendAgainstThreats(const Update& event)
     setAlignment(event, crease_defender_assignments, TbotsProto::BallStealMode::STEAL);
     updatePassDefenderControlParams(pass_defender_assignments,
                                     TbotsProto::BallStealMode::STEAL);
+}
+
+void DefensePlayFSM::updateShadowers(const Update& event,
+                                     const std::vector<EnemyThreat>& threats_to_shadow)
+{
+    setUpShadowers(static_cast<unsigned int>(threats_to_shadow.size()));
+
+    for (unsigned int i = 0; i < shadowers.size(); i++)
+    {
+        shadowers.at(i)->updateControlParams(threats_to_shadow.at(i),
+                                             ROBOT_SHADOWING_DISTANCE_METERS);
+    }
+}
+
+
+void DefensePlayFSM::setUpShadowers(int num_shadowers)
+{
+    if (num_shadowers == (int)shadowers.size())
```

Comment:
we don't use C-style casts
```suggestion
    if (num_shadowers == static_cast<int>(shadowers.size()))
```